### ISSUE TITLE
Branch clean-up

### ISSUE THREAD

**Issue by Elliria** [autokey/autokey#810]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

None

### Choose one or more terms that describe this issue:

- [ ] autokey triggers
- [ ] autokey-gtk
- [ ] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [X] development
- [ ] documentation
- [ ] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [X] technical debt
- [ ] user interface

### Which AutoKey GUI did you use?

None

### Can you briefly describe the issue?

There are some branches in the AutoKey source code that could use investigation and possible removal.

### Can the issue be reproduced?

None

### What are the steps to reproduce the issue?

Examine the [branches](https://github.com/autokey/autokey/branches) in the AutoKey project.

### What should have happened?

Branches that are no longer needed should have been deleted.

### What actually happened?

There are some branches in the AutoKey project that may or may not still be needed. Namely:
* [bugfix/issue_291](https://github.com/autokey/autokey/tree/bugfix/issue_291)
* [bugfix/issue_334](https://github.com/autokey/autokey/tree/bugfix/issue_334)
* [revert-538-develop-docupdate](https://github.com/autokey/autokey/tree/revert-538-develop-docupdate)

**Comment by Elliria** [comment]
[Issue #291](https://github.com/autokey/autokey/issues/291) has been completed and closed and there's a link to [1c5e3ce](https://github.com/autokey/autokey/commit/1c5e3ce8b45fbea49fd17c2b358211434605a89a) near the bottom of the issue comments. What worries me is that when I examine [the branch](https://github.com/autokey/autokey/tree/bugfix/issue_291), it's [4 commits ahead](https://github.com/autokey/autokey/compare/master...bugfix/issue_291) of **master**. Hopefully @luziferius can take a look at it and see if something needs to be done with it or if it can be deleted.

**Comment by Elliria** [comment]
I haven't looked into the others yet.

**Comment by josephj11** [comment]
#291 - closed as completed - I think this was about adding the ignore capslock feature. Branch should be safe to delete.

#334 - open - quite technical - requires expertise in Python and AutoKey - we'd have to do a diff between this branch and master to see if there's good work in it that cleans up Autokey and gets rid of some xfails. But, I'm not sure if any of us currently know enough to safely apply any such fixes.

#538 was all @sebastiansam55 and @BlueDrink9 working on sphinx documentation. I believe this is the change that was reverted when it wasn't completed in 2021. I expect Sam's current work to get it functional will make this obsolete so we can delete it, but not without his approval.

@luziferius ghosted us more than a year ago. Looks like he dropped off the map entirely. It looks like he did a couple of commits in other projects in February and March, so maybe he's not totally gone, but he hasn't answered any of our messages. It's pretty strange because, AFAIK, there were no conflicts or even disagreements, etc. Everybody was happy.

**Comment by josephj11** [comment]
I looked at your link and saw a bunch of GTK code around adding Ignore CapsLock to the GUI. How do I see the 4 commits that are ahead of master? I probably won't know what I'm looking at, but I'll give it a go once I can see them.

**Comment by Elliria** [comment]
Does this link show them to you? [https://github.com/autokey/autokey/compare/master...bugfix/issue_291](https://github.com/autokey/autokey/compare/master...bugfix/issue_291)

**Comment by Elliria** [comment]
It's possible that it's perfectly fine for the branch to be ahead of **master** if it was branched off of **develop** and if that's where those commits were merged into. If that's the case, then that branch can be deleted. I don't yet know how to tell that, though, and must go educate myself.

**Comment by Elliria** [comment]
Okay, I can see that #334 and #538 still need work, although I get the impression that #334 is in reference to legacy code from Python 2. Either @luziferius or @BlueDrink9 would know, so hopefully one of them will see this and jump in.

**Comment by josephj11** [comment]
This branch started out as Python 3. That was its reason for being. Of course, it started out with all the Python 2 code, so some may be left in it if it didn't cause syntax errors, etc.

> Does this link show them to you? [master...bugfix/issue_291](https://github.com/autokey/autokey/compare/master...bugfix/issue_291)

Am I correct in thinking that green signifies things that are in the test branch, but not in the master?
That can't be right because all the "good stuff" is in green and that functionality works in 0.96.0 (and, probably, before that).

If the red is the "new" code that is only in the test branch, that all looks like it should be deleted and if that's all the changes, then that branch can definitely be deleted.

**Comment by Elliria** [comment]
Nope. In a comparison, the branches at the top signify where the merge wants to happen **from** (the **compare** branch) and **to** (the **base** branch. In [this comparison](https://github.com/autokey/autokey/compare/master...bugfix/issue_291), the **bugfix/issue_29** branch wanted to do a merge into the **master** branch. The listed commits contain the changes that were done to files in the **bugfix/issue_29** branch. Beneath that list are the diffs that show what changed in each file in those commits. In the diffs, the green background signifies something that was added, the red background signifies something that was removed, and the white background is for surrounding text that didn't change and is just there for context, I suppose.

I also appreciate the little indicators that are displayed on diffs, commits, and PRs and show you at a glance what happened with color-coding:

![QuickOverview](https://user-images.githubusercontent.com/25792925/224718968-a2be94d7-af2e-41cc-acbb-a18db194bc79.png)

**Comment by Elliria** [comment]
It looks like **bugfix/issue_291** was the only one that needed to be removed and it's gone, so this issue is solved.
